include/gul14/to_number.h

```diff
@@ -563,6 +565,20 @@ constexpr inline optional<NumberType> to_number(gul14::string_view str) noexcept
     return detail::to_unsigned_float<NumberType>(str);
 }
 
+template<typename NumberType, std::enable_if_t<std::is_same<NumberType, bool>::value, int> = 0>
+constexpr inline optional<NumberType> to_number(gul14::string_view str) noexcept
+{
+    constexpr std::array<gul14::string_view const, 2> yes = { "1", "true" };
+    if (std::find(yes.cbegin(), yes.cend(), gul14::lowercase_ascii(str)) != yes.cend())
+            return { true };
+
+    constexpr std::array<gul14::string_view const, 2> no = { "0", "false" };
+    if (std::find(no.cbegin(), no.cend(), gul14::lowercase_ascii(str)) != no.cend())
+            return { false };
+
+    return {};
+}
```

Comment:
In a way, this is very elegant, but it seems to me that we are leaving performance on the table.
- If we throw a long string at the function, it is converted to lowercase twice before we realize that we cannot do anything with it.
- I would assume that "1" and "0" are the default representations that we should optimize for (all other `to_number` overloads deal primarily with digits as well).

So maybe a more low-level implementation would be in order, along the lines of:
```cpp
if (str == "1")
    return true;
else if (str == "0")
    return false;
else if (equals_nocase(str, "true"))
    return true;
else if (equals_nocase(str, "false"))
    return false;
else
    return {};
```
Or even, probably with the minimum amount of work:
```cpp
switch (str.length())
{
case 1:
    if (str.front() == '0')
        return false;
    else if (str.front() == '1')
        return true;
    break;
case 4:
    if (gul14::lowercase_ascii(str) == "true")
        return true;
    break;
case 5:
    if (gul14::lowercase_ascii(str) == "false")
        return false;
    break;
};
return {};
```

Reply:
I would call that premature optimization.
(But I must confess I like the `length()` approach quite a lot!)